homeassistant/components/iron_os/__init__.py

```diff
@@ -39,10 +51,19 @@ async def async_setup_entry(hass: HomeAssistant, entry: IronOSConfigEntry) -> bo
 
     device = Pynecil(ble_device)
 
-    coordinator = IronOSCoordinator(hass, device)
+    coordinator = IronOSLiveDataCoordinator(hass, device)
     await coordinator.async_config_entry_first_refresh()
 
-    entry.runtime_data = coordinator
+    session = async_get_clientsession(hass)
+    github = GitHubAPI(session=session)
+
+    firmware_update_coordinator = IronOSFirmwareUpdateCoordinator(hass, device, github)
+    await firmware_update_coordinator.async_config_entry_first_refresh()
```

Comment:
Would it make sense to have this set in `async_setup` and added to `hass.data` just like in `wled`? I know that it's more likely to have multiple `wled` devices than multiple irons, but I like to think that its nice to see it as a good practice.

WDYT? (not seeing this as a blocker btw, just thinking out loud)

Reply:
You mean one single coordinator for all instead of per entry? Hmm, I think that makes sense but if it is not a blocker, I would prefer to do this in a follow-up PR

Reply:
Fine by me